From be102289d381a1c8f2a5eccdc1bba7bdf7593baa Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 21 Jun 2012 09:45:06 -0400
Subject: [PATCH] Fix decl of pe_update off_t -> loff_t

---
 libdpe/pe_update.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdpe/pe_update.c b/libdpe/pe_update.c
index ca65bca..15678f6 100644
--- a/libdpe/pe_update.c
+++ b/libdpe/pe_update.c
@@ -75,7 +75,7 @@ write_file(Pe *pe, off_t size, size_t shnum)
 	return size;
 }
 
-off_t
+loff_t
 pe_update(Pe *pe, Pe_Cmd cmd)
 {
 	if (cmd != PE_C_NULL && cmd != PE_C_WRITE && cmd != PE_C_WRITE_MMAP) {
-- 
1.7.10.2

