From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Robbie Harwood <rharwood@redhat.com>
Date: Tue, 8 Mar 2022 16:30:01 -0500
Subject: [PATCH] Disable wchar size warning (for armv7hl)

Signed-off-by: Robbie Harwood <rharwood@redhat.com>
---
 Make.defaults | 4 ++++
 util/Makefile | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/Make.defaults b/Make.defaults
index 130c1ee..1724caa 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -67,6 +67,10 @@ cflags	= $(CFLAGS) $(ARCH3264) \
 	$(if $(filter $(CC),clang),$(clang_cflags), ) \
 	$(if $(filter $(CC),gcc),$(gcc_cflags), ) \
 	$(call pkg-config-cflags)
+ifeq ($(ARCH), armv7l)
+	cflags += -Wl,--no-wchar-size-warning
+endif
+
 clang_ccldflags =
 gcc_ccldflags = -fno-merge-constants \
 	-fvar-tracking -fvar-tracking-assignments -fkeep-inline-functions \
diff --git a/util/Makefile b/util/Makefile
index 96463ba..4897299 100644
--- a/util/Makefile
+++ b/util/Makefile
@@ -11,6 +11,9 @@ LDFLAGS = -nostdlib -T $(LIBDIR)/gnuefi/elf_$(HOSTARCH)_efi.lds -shared -Bsymbol
 LIBS=-lefi -lgnuefi $(shell $(CC) -print-libgcc-file-name)
 CCLDFLAGS =
 BUILDFLAGS = -I/usr/include/efi/ -I/usr/include/efi/$(HOSTARCH)/ -I/usr/include/efi/protocol -fpic -fshort-wchar -fno-reorder-functions -fno-strict-aliasing -fno-merge-constants -mno-red-zone -Wimplicit-function-declaration
+ifeq ($(ARCH, armv7l)
+	BUILDFLAGS += -Wl,--no-wchar-size-warning
+endif
 
 TARGETS = setupsb.efi dumpsb.efi clearpk.efi
 
