From 39466ae9ed3ce5f78fc20c6e74eb0fb3aa93349e Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 6 Aug 2013 16:49:06 -0400
Subject: [PATCH 5/9] Try harder to figure out if this is RHEL.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/macros.pesign | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/macros.pesign b/src/macros.pesign
index 244f576..f94553d 100644
--- a/src/macros.pesign
+++ b/src/macros.pesign
@@ -22,7 +22,7 @@
 # -s 					# perform signing
 %pesign(i:o:C:e:c:n:a:s)						\
   if [ -x %{_pesign} -a "%{_target_cpu}" == "x86_64" ]; then		\
-    if [ -e /etc/rhel-release ]; then					\
+    if [ "0%{?rhel}" -ge "7" ]; then					\
       nss=$(mktemp -p $PWD -d)						\
       certutil -d ${nss} -N						\
       certutil -A -n "ca" -t "CT,C," -i %{-a*} -a -d ${nss}		\
-- 
1.8.3.1

