From f8b19278775fe8a5c599b94fcae90b99a781a42b Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 7 Aug 2013 09:06:33 -0400
Subject: [PATCH 6/8] Don't use ASCII mode for RHEL certificate imports.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/macros.pesign | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/macros.pesign b/src/macros.pesign
index f94553d..84e87a3 100644
--- a/src/macros.pesign
+++ b/src/macros.pesign
@@ -25,8 +25,8 @@
     if [ "0%{?rhel}" -ge "7" ]; then					\
       nss=$(mktemp -p $PWD -d)						\
       certutil -d ${nss} -N						\
-      certutil -A -n "ca" -t "CT,C," -i %{-a*} -a -d ${nss}		\
-      certutil -A -n "signer" -t ",c," -i %{-c*} -a -d ${nss}		\
+      certutil -A -n "ca" -t "CT,C," -i %{-a*} -d ${nss}		\
+      certutil -A -n "signer" -t ",c," -i %{-c*} -d ${nss}		\
       sattrs=$(mktemp -p $PWD --suffix=.der)				\
       %{_pesign} %{-i} -E ${sattrs} --certdir ${nss}			\
       rpm-sign --key "%{-n*}" --rsasign ${sattrs}			\
-- 
1.8.3.1

