From 682233d107460b49071017b4d88c0430373dbd35 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 10:55:25 -0400
Subject: [PATCH 06/30] Update valgrind.supp for newer codepaths.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/valgrind.supp | 361 ++++++++++++++++++++++++++++++++++++------------------
 1 file changed, 245 insertions(+), 116 deletions(-)

diff --git a/src/valgrind.supp b/src/valgrind.supp
index 3a51293..0a8edd9 100644
--- a/src/valgrind.supp
+++ b/src/valgrind.supp
@@ -1,23 +1,43 @@
 {
    <insert_a_suppression_name_here>
-   Memcheck:Leak
-   fun:malloc
-   fun:NSSRWLock_New_Util
-   fun:NSSTrustDomain_Create
-   fun:STAN_LoadDefaultNSS3TrustDomain
-   fun:nss_Init
-   fun:NSS_Init
-   fun:main
+   Memcheck:Param
+   socketcall.sendto(msg)
+   fun:send
+   fun:MessageSend
+   fun:MessageSendWithHeader
+   fun:SCardConnect
+   fun:CKYCardConnection_Connect
+   fun:_ZN4Slot14connectToTokenEv
+   fun:_ZN4Slot17refreshTokenStateEv
+   fun:_ZN4Slot14isTokenPresentEv
+   fun:_ZN8SlotList11getSlotListEhPmS0_
+   fun:C_GetSlotList
+   fun:secmod_LoadPKCS11Module
+   fun:SECMOD_LoadModule
 }
 {
    <insert_a_suppression_name_here>
-   Memcheck:Leak
-   fun:_Znwm
-   obj:*
-   fun:secmod_ModuleInit
+   Memcheck:Cond
+   fun:SCardStatus
+   fun:CKYCardConnection_GetStatus
+   fun:_ZN4Slot23cardStateMayHaveChangedEv
+   fun:_ZN4Slot17refreshTokenStateEv
+   fun:_ZN4Slot14isTokenPresentEv
+   fun:_ZN8SlotList11getSlotListEhPmS0_
+   fun:C_GetSlotList
    fun:secmod_LoadPKCS11Module
    fun:SECMOD_LoadModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
+   fun:nss_Init
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:malloc
+   fun:NSSRWLock_New_Util
+   fun:NSSTrustDomain_Create
+   fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -25,12 +45,15 @@
 {
    <insert_a_suppression_name_here>
    Memcheck:Leak
-   fun:_Znwm
-   obj:*
-   fun:secmod_ModuleInit
+   fun:malloc
+   fun:PORT_Alloc_Util
+   fun:PK11_ReadMechanismList
+   fun:PK11_InitToken
+   fun:PK11_InitSlot
    fun:secmod_LoadPKCS11Module
    fun:SECMOD_LoadModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -223,12 +246,10 @@
 {
    <insert_a_suppression_name_here>
    Memcheck:Leak
-   fun:_Znwm
-   obj:*
-   fun:secmod_ModuleInit
-   fun:secmod_LoadPKCS11Module
-   fun:SECMOD_LoadModule
-   fun:SECMOD_LoadModule
+   fun:calloc
+   fun:NSSRWLock_New_Util
+   fun:NSSTrustDomain_Create
+   fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -237,10 +258,10 @@
    <insert_a_suppression_name_here>
    Memcheck:Leak
    fun:malloc
-   fun:PKIX_PL_Malloc
-   fun:PKIX_PL_Object_Alloc
-   fun:PKIX_PL_HashTable_Create
-   fun:PKIX_Initialize
+   fun:PR_NewCondVar
+   fun:NSSRWLock_New_Util
+   fun:NSSTrustDomain_Create
+   fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -249,18 +270,7 @@
    <insert_a_suppression_name_here>
    Memcheck:Leak
    fun:malloc
-   fun:PKIX_PL_Malloc
-   fun:PKIX_PL_Object_Alloc
-   fun:PKIX_PL_HashTable_Create
-   fun:PKIX_Initialize
-   fun:nss_Init
-   fun:NSS_Init
-   fun:main
-}
-{
-   <insert_a_suppression_name_here>
-   Memcheck:Leak
-   fun:calloc
+   fun:PR_NewCondVar
    fun:NSSRWLock_New_Util
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -272,10 +282,10 @@
    <insert_a_suppression_name_here>
    Memcheck:Leak
    fun:malloc
-   fun:PR_NewCondVar
-   fun:NSSRWLock_New_Util
-   fun:NSSTrustDomain_Create
-   fun:STAN_LoadDefaultNSS3TrustDomain
+   fun:PKIX_PL_Malloc
+   fun:PKIX_PL_Object_Alloc
+   fun:PKIX_PL_HashTable_Create
+   fun:PKIX_Initialize
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -284,10 +294,10 @@
    <insert_a_suppression_name_here>
    Memcheck:Leak
    fun:malloc
-   fun:PR_NewCondVar
-   fun:NSSRWLock_New_Util
-   fun:NSSTrustDomain_Create
-   fun:STAN_LoadDefaultNSS3TrustDomain
+   fun:PKIX_PL_Malloc
+   fun:PKIX_PL_Object_Alloc
+   fun:PKIX_PL_HashTable_Create
+   fun:PKIX_Initialize
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -357,8 +367,10 @@
    fun:calloc
    fun:PORT_ZAlloc_Util
    fun:PORT_NewArena_Util
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -368,8 +380,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateCertificate
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -382,8 +396,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateItem
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -396,8 +412,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateString
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -410,8 +428,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateString
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -424,28 +444,32 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateCertificate
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
-   fun:NSS_Init
-   fun:main
 }
 {
    <insert_a_suppression_name_here>
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateItem
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
-   fun:NSS_Init
-   fun:main
 }
 {
    <insert_a_suppression_name_here>
@@ -466,6 +490,7 @@
    fun:calloc
    fun:nss_ZAlloc
    fun:nssArena_Create
+   fun:NSSArena_Create
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
@@ -505,7 +530,10 @@
    fun:calloc
    fun:nss_ZAlloc
    fun:nssArena_Create
+   fun:NSSArena_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -514,14 +542,12 @@
 {
    <insert_a_suppression_name_here>
    Memcheck:Leak
-   fun:malloc
-   obj:*
-   obj:*
-   obj:*
-   fun:secmod_ModuleInit
-   fun:secmod_LoadPKCS11Module
-   fun:SECMOD_LoadModule
-   fun:SECMOD_LoadModule
+   fun:calloc
+   fun:nss_ZAlloc
+   fun:nssTokenObjectCache_Create
+   fun:nssToken_CreateFromPK11SlotInfo
+   fun:STAN_InitTokenForSlotInfo
+   fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -531,28 +557,32 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateCertificate
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
-   fun:NSS_Init
-   fun:main
 }
 {
    <insert_a_suppression_name_here>
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateItem
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
-   fun:NSS_Init
-   fun:main
 }
 {
    <insert_a_suppression_name_here>
@@ -560,35 +590,11 @@
    fun:calloc
    fun:PORT_ZAlloc_Util
    fun:PORT_NewArena_Util
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
    fun:SECMOD_LoadModule
-   fun:nss_Init
-   fun:NSS_Init
-   fun:main
-}
-{
-   <insert_a_suppression_name_here>
-   Memcheck:Leak
-   fun:calloc
-   fun:nss_ZAlloc
-   fun:nssArena_Create
-   fun:nssToken_CreateFromPK11SlotInfo
-   fun:STAN_InitTokenForSlotInfo
-   fun:STAN_LoadDefaultNSS3TrustDomain
-   fun:nss_Init
-   fun:NSS_Init
-   fun:main
-}
-{
-   <insert_a_suppression_name_here>
-   Memcheck:Leak
-   fun:calloc
-   fun:nss_ZAlloc
-   fun:nssArena_Create
-   fun:nssToken_CreateFromPK11SlotInfo
-   fun:STAN_InitTokenForSlotInfo
-   fun:STAN_LoadDefaultNSS3TrustDomain
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -599,8 +605,10 @@
    fun:calloc
    fun:PR_NewLock
    fun:PORT_NewArena_Util
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -610,8 +618,10 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -621,22 +631,11 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:PKIX_PL_Object_Alloc
-   fun:PKIX_PL_HashTable_Create
-   fun:PKIX_Initialize
-   fun:nss_Init
-   fun:NSS_Init
-   fun:main
-}
-{
-   <insert_a_suppression_name_here>
-   Memcheck:Leak
-   fun:calloc
-   fun:PR_NewLock
-   fun:PKIX_PL_Object_Alloc
-   fun:PKIX_PL_Mutex_Create
-   fun:PKIX_PL_HashTable_Create
-   fun:PKIX_Initialize
+   fun:PK11_NewSlotInfo
+   fun:secmod_LoadPKCS11Module
+   fun:SECMOD_LoadModule
+   fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -646,9 +645,11 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:PKIX_PL_Mutex_Create
-   fun:PKIX_PL_HashTable_Create
-   fun:PKIX_Initialize
+   fun:PK11_NewSlotInfo
+   fun:secmod_LoadPKCS11Module
+   fun:SECMOD_LoadModule
+   fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -658,9 +659,10 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:PKIX_PL_Object_Alloc
-   fun:PKIX_PL_HashTable_Create
-   fun:PKIX_Initialize
+   fun:nssArena_Create
+   fun:NSSArena_Create
+   fun:NSSTrustDomain_Create
+   fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -670,7 +672,7 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:nssArena_Create
+   fun:NSSRWLock_New_Util
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
@@ -682,7 +684,8 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:NSSRWLock_New_Util
+   fun:nssArena_Create
+   fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
@@ -694,7 +697,6 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:nssArena_Create
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -707,6 +709,8 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
+   fun:nssHash_Create
+   fun:nssHash_CreateCertificate
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -720,6 +724,7 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssHash_Create
+   fun:nssHash_CreateItem
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -733,6 +738,7 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssHash_Create
+   fun:nssHash_CreateString
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -746,6 +752,7 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssHash_Create
+   fun:nssHash_CreateString
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -758,9 +765,7 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:nssHash_Create
-   fun:nssTrustDomain_InitializeCache
-   fun:NSSTrustDomain_Create
+   fun:nssList_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -771,7 +776,9 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
-   fun:nssList_Create
+   fun:nssTokenObjectCache_Create
+   fun:nssToken_CreateFromPK11SlotInfo
+   fun:STAN_InitTokenForSlotInfo
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -821,7 +828,10 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssArena_Create
+   fun:NSSArena_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -834,6 +844,8 @@
    fun:PR_NewLock
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -845,8 +857,11 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssHash_Create
+   fun:nssHash_CreateCertificate
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -858,8 +873,11 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssHash_Create
+   fun:nssHash_CreateItem
    fun:nssCertificateStore_Create
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -871,6 +889,55 @@
    fun:calloc
    fun:PR_NewLock
    fun:PKIX_PL_Object_Alloc
+   fun:PKIX_PL_HashTable_Create
+   fun:PKIX_Initialize
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:calloc
+   fun:PR_NewLock
+   fun:PKIX_PL_Object_Alloc
+   fun:PKIX_PL_Mutex_Create
+   fun:PKIX_PL_HashTable_Create
+   fun:PKIX_Initialize
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:calloc
+   fun:PR_NewLock
+   fun:PKIX_PL_Mutex_Create
+   fun:PKIX_PL_HashTable_Create
+   fun:PKIX_Initialize
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:calloc
+   fun:PR_NewLock
+   fun:PKIX_PL_Object_Alloc
+   fun:PKIX_PL_HashTable_Create
+   fun:PKIX_Initialize
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:calloc
+   fun:PR_NewLock
+   fun:PKIX_PL_Object_Alloc
    fun:PKIX_PL_Mutex_Create
    fun:PKIX_PL_HashTable_Create
    fun:PKIX_Initialize
@@ -1064,6 +1131,33 @@
 {
    <insert_a_suppression_name_here>
    Memcheck:Leak
+   fun:calloc
+   fun:nss_ZAlloc
+   fun:nssArena_Create
+   fun:nssToken_CreateFromPK11SlotInfo
+   fun:STAN_InitTokenForSlotInfo
+   fun:STAN_LoadDefaultNSS3TrustDomain
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:calloc
+   fun:nss_ZAlloc
+   fun:nssArena_Create
+   fun:nssSlot_CreateFromPK11SlotInfo
+   fun:nssToken_CreateFromPK11SlotInfo
+   fun:STAN_InitTokenForSlotInfo
+   fun:STAN_LoadDefaultNSS3TrustDomain
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
    fun:malloc
    fun:PKIX_PL_Malloc
    fun:pkix_pl_PrimHashTable_Create
@@ -1126,8 +1220,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateCertificate
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -1140,8 +1236,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateItem
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -1154,8 +1252,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateString
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -1168,8 +1268,10 @@
    Memcheck:Leak
    fun:calloc
    fun:nss_ZAlloc
+   fun:nss_arena_hash_alloc_table
    fun:PL_NewHashTable
    fun:nssHash_Create
+   fun:nssHash_CreateString
    fun:nssTrustDomain_InitializeCache
    fun:NSSTrustDomain_Create
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -1183,9 +1285,11 @@
    fun:calloc
    fun:PR_NewLock
    fun:PORT_NewArena_Util
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -1195,9 +1299,11 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -1233,6 +1339,7 @@
    fun:calloc
    fun:PR_NewLock
    fun:nssArena_Create
+   fun:nssSlot_CreateFromPK11SlotInfo
    fun:nssToken_CreateFromPK11SlotInfo
    fun:STAN_InitTokenForSlotInfo
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -1245,6 +1352,7 @@
    Memcheck:Leak
    fun:calloc
    fun:PR_NewLock
+   fun:nssSlot_CreateFromPK11SlotInfo
    fun:nssToken_CreateFromPK11SlotInfo
    fun:STAN_InitTokenForSlotInfo
    fun:STAN_LoadDefaultNSS3TrustDomain
@@ -1258,8 +1366,24 @@
    fun:malloc
    fun:PL_ArenaAllocate
    fun:PORT_ArenaAlloc_Util
+   fun:secmod_NewModule
    fun:SECMOD_CreateModule
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
+   fun:nss_Init
+   fun:NSS_Init
+   fun:main
+}
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:malloc
+   fun:PORT_Alloc_Util
+   fun:PK11_NewSlotInfo
+   fun:secmod_LoadPKCS11Module
+   fun:SECMOD_LoadModule
+   fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -1273,7 +1397,9 @@
    fun:legacy_ReadSecmodDB
    fun:sftkdb_ReadSecmodDB
    fun:NSC_ModuleDBFunc
+   fun:SECMOD_GetModuleSpecList
    fun:SECMOD_LoadModule
+   fun:nss_InitModules
    fun:nss_Init
    fun:NSS_Init
    fun:main
@@ -1349,6 +1475,8 @@
    fun:nss_zalloc_arena_locked
    fun:nss_ZAlloc
    fun:nssCryptoContext_Create
+   fun:nssTrustDomain_CreateCryptoContext
+   fun:NSSTrustDomain_CreateCryptoContext
    fun:STAN_LoadDefaultNSS3TrustDomain
    fun:nss_Init
    fun:NSS_Init
@@ -1375,6 +1503,7 @@
    fun:PL_ArenaAllocate
    fun:nss_zalloc_arena_locked
    fun:nss_ZAlloc
+   fun:nssSlot_CreateFromPK11SlotInfo
    fun:nssToken_CreateFromPK11SlotInfo
    fun:STAN_InitTokenForSlotInfo
    fun:STAN_LoadDefaultNSS3TrustDomain
-- 
1.7.12.1

