From 50c50c8fbebab3d8b5efff35dc1a7ca4b44d6b19 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 11:08:30 -0400
Subject: [PATCH 08/30] [valgrind] Don't complain about unlocking a key and
 keeping the handle.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/valgrind.supp | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/valgrind.supp b/src/valgrind.supp
index 0a8edd9..62934e8 100644
--- a/src/valgrind.supp
+++ b/src/valgrind.supp
@@ -1511,3 +1511,19 @@
    fun:NSS_Init
    fun:main
 }
+{
+   <insert_a_suppression_name_here>
+   Memcheck:Leak
+   fun:calloc
+   fun:nss_ZAlloc
+   fun:nssArena_Create
+   fun:nssList_Create
+   fun:nssTrustDomain_UpdateCachedTokenCerts
+   fun:PK11_DoPassword
+   fun:PK11_Authenticate
+   fun:unlock_nss_token
+   fun:handle_unlock_token
+   fun:handle_event
+   fun:handle_events
+   fun:daemonize
+}
-- 
1.7.12.1

