From 5b8950a8cddad1076fb631c4ef6999bfb4f977f8 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 7 Aug 2013 09:37:33 -0400
Subject: [PATCH 8/9] Use --force when we've got a sattrs blob from mktemp()

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/macros.pesign | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/macros.pesign b/src/macros.pesign
index 6b22826..a0339fe 100644
--- a/src/macros.pesign
+++ b/src/macros.pesign
@@ -28,7 +28,7 @@
       certutil -A -n "ca" -t "CT,C," -i %{-a*} -d ${nss}		\
       certutil -A -n "signer" -t ",c," -i %{-c*} -d ${nss}		\
       sattrs=$(mktemp -p $PWD --suffix=.der)				\
-      %{_pesign} %{-i} -E ${sattrs} --certdir ${nss}			\
+      %{_pesign} %{-i} -E ${sattrs} --certdir ${nss} --force		\
       rpm-sign --key "%{-n*}" --rsasign ${sattrs}			\
       %{_pesign} -R ${sattrs}.sig -I ${sattrs} %{-i}			\\\
                  --certdir ${nss} -c signer %{-o}			\
-- 
1.8.3.1

