From f966137c17f74fc3e343dfb6e04300a9d179de03 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 12:05:29 -0400
Subject: [PATCH 10/36] Check for NSS_Shutdown() failure.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/daemon.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/daemon.c b/src/daemon.c
index 245491f..77fff90 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -967,6 +967,11 @@ daemonize(cms_context *cms_ctx, int do_fork)
 
 	rc = handle_events(&ctx);
 
-	NSS_Shutdown();
+	status = NSS_Shutdown();
+	if (status != SECSuccess) {
+		fprintf(stderr, "NSS_Shutdown failed: %s\n",
+			PORT_ErrorToString(PORT_GetError()));
+		exit(1);
+	}
 	return rc;
 }
-- 
1.7.12.1

