From c196b462ad5267e8ed20c0b855b9921268b22a7b Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 15:26:47 -0400
Subject: [PATCH 18/42] Minor whitespace fixes.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/daemon.c | 6 +++---
 src/pesign.c | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/daemon.c b/src/daemon.c
index 73fd924..69aea4b 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -70,7 +70,7 @@ hide_stolen_goods_from_cms(cms_context *new, cms_context *old)
 
 static void
 send_response(context *ctx, cms_context *cms, struct pollfd *pollfd, int rc)
-{	
+{
 	struct msghdr msg;
 	struct iovec iov;
 	ssize_t n;
@@ -890,7 +890,7 @@ int
 daemonize(cms_context *cms_ctx, int do_fork)
 {
 	int rc = 0;
-	context ctx = { 
+	context ctx = {
 		.backup_cms = cms_ctx,
 		.priority = do_fork ? LOG_PID
 				    : LOG_PID|LOG_PERROR,
@@ -904,7 +904,7 @@ daemonize(cms_context *cms_ctx, int do_fork)
 		fprintf(stderr, "pesignd must be started as root");
 		exit(1);
 	}
-		
+
 	check_socket(&ctx);
 
 	openlog("pesignd", LOG_PID, LOG_DAEMON);
diff --git a/src/pesign.c b/src/pesign.c
index 819cee0..2ba5ef1 100644
--- a/src/pesign.c
+++ b/src/pesign.c
@@ -88,7 +88,7 @@ open_input(pesign_context *ctx)
 
 	struct stat statbuf;
 	ctx->infd = open(ctx->infile, O_RDONLY|O_CLOEXEC);
-	stat(ctx->infile, &statbuf); 
+	stat(ctx->infile, &statbuf);
 	ctx->outmode = statbuf.st_mode;
 
 	if (ctx->infd < 0) {
-- 
1.7.12.1

