From a1ce809e199c7fbbd6f5c0e75f27a4234fcbd2bc Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 15:34:00 -0400
Subject: [PATCH 22/30] Handle errors on pesign_context_init()

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/pesign_context.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/pesign_context.c b/src/pesign_context.c
index c6afda6..cbd929f 100644
--- a/src/pesign_context.c
+++ b/src/pesign_context.c
@@ -37,7 +37,9 @@ pesign_context_new(pesign_context **ctx)
 	if (!context)
 		return -1;
 
-	pesign_context_init(context);
+	rc = pesign_context_init(context);
+	if (rc < 0)
+		return rc;
 	context->flags |= PESIGN_C_ALLOCATED;
 
 	*ctx = context;
-- 
1.7.12.1

