From 24a63eab7ddbe2be3ab6b25b04602d8e3fe5d775 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 18 Oct 2012 14:28:36 -0400
Subject: [PATCH 33/36] Allow use of -e from rpm macro.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/macros.pesign | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/macros.pesign b/src/macros.pesign
index 703edbb..7706050 100644
--- a/src/macros.pesign
+++ b/src/macros.pesign
@@ -11,9 +11,9 @@
 
 %_pesign /usr/bin/pesign
 
-%pesign(i:o:C:s) \
+%pesign(i:o:C:e:s) \
   if [ -x %{_pesign} -a "%{_target_cpu}" == "x86_64" ]; then \
-    %{_pesign} %{__pesign_token} %{__pesign_cert} %{-i} %{-o} %{-s} \
+    %{_pesign} %{__pesign_token} %{__pesign_cert} %{-i} %{-o} %{-e} %{-s} \
   else \
     if [ -n "%{-i*}" -a -n "%{-o*}" ]; then \
       mv %{-i*} %{-o*} \
-- 
1.7.12.1

