From 9c2daa8d3761b49961498cb9a9bbc8a37e05b0da Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 19 Oct 2012 10:19:39 -0400
Subject: [PATCH 40/41] Use setfacl in sysvinit script to allow kojibuilder
 access.

---
 src/pesign.sysvinit | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pesign.sysvinit b/src/pesign.sysvinit
index f955e01..ea37c58 100644
--- a/src/pesign.sysvinit
+++ b/src/pesign.sysvinit
@@ -24,6 +24,10 @@ start(){
     RETVAL=$?
     echo
     touch /var/lock/subsys/pesign
+    setfacl -m u:kojibuilder:x /var/run/pesign
+    setfacl -m u:kojibuilder:rw /var/run/pesign/socket
+    setfacl -m g:kojibuilder:x /var/run/pesign
+    setfacl -m g:kojibuilder:rw /var/run/pesign/socket
 }
 
 stop(){
-- 
1.7.12.1

