From a967a147079085fce9b3a4d66cbdd28ccbab559f Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 28 Sep 2012 14:27:54 -0400
Subject: [PATCH] Fix missing section relocation when we've added space.

---
 libdpe/pe_allocspace.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libdpe/pe_allocspace.c b/libdpe/pe_allocspace.c
index d766503..0ae1f5d 100644
--- a/libdpe/pe_allocspace.c
+++ b/libdpe/pe_allocspace.c
@@ -46,6 +46,7 @@ pe_fix_addresses(Pe *pe, int64_t offset)
 			&pe->state.pe.shdr[cnt];
 
 		adjust(pe->state.pe.scns.data[cnt].rawdata_base, offset);
+		adjust(pe->state.pe.scns.data[cnt].data_base, offset);
 	}
 }
 #undef adjust
-- 
1.7.12.1

