From a8e0e93a15ae57dcfc2bf20921fc5fe604dee3fa Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 14 Aug 2012 15:44:26 -0400
Subject: [PATCH] Only sign things on x86_64.

---
 src/macros.pesign | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/macros.pesign b/src/macros.pesign
index beb6489..7932629 100644
--- a/src/macros.pesign
+++ b/src/macros.pesign
@@ -9,6 +9,6 @@
 %__pesign_token %{expand: %%{nil} %%{?pe_signing_token:-t "%%{pe_signing_token}"}}
 %__pesign_cert %{expand: %%{!?pe_signing_cert:-c "Red Hat Test Certificate"} %%{?pe_signing_cert:-c "%%{pe_signing_cert}"}}
 
-%_pesign /usr/bin/pesign
-%pesign \
-%{_pesign} %{__pesign_token} %{__pesign_cert}
+%_pesign %{expand:%%([ %%{_arch} == x86_64 ] && echo /usr/bin/pesign || echo /usr/bin/true)}
+
+%pesign %{_pesign} %{__pesign_token} %{__pesign_cert}
-- 
1.7.11.4

